### ISSUE TITLE
Version 5 core components discussion

### ISSUE THREAD

**Issue by thomasthesecond** [lonelyplanet/backpack-ui#618]
In version 5, I would like to remove most of the components that are currently in version 4. The thinking here is that backpack-ui should contain the core building blocks for the UI. If we're talking about atomic design, then backpack-ui should contain mostly atoms and maybe some molecules that are reusable.

I did a quick audit of our UI and here's a list of components I think we can start with:

* ad
* avatar
* button
* checkbox
* container
* dialog / modal
* divider (would replace `border-bottom` on a lot of things)
* grid (container for CSS grid layout)
* heading
* icon
* iconButton
* image
* input
* moreLink
* panel (maybe just a box with drop shadow? we have some UI that could fit this)
* radio
* shareButton
* text (we have a handful of text components now, not sure if we should keep them grouped)
* textarea
* toast

There are probably some molecule-type components that could be built with the atoms above and be kept in backpack-ui, like a shareMenu maybe, but I think first we figure out what the atoms are.

Are there any I missed? Any thoughts about this approach?

**Comment by nickfranciosi** [comment]
approach sounds good. I think we have a lot of repetition currently that could be broken down to these smaller pieces. 

As far as thinking through all the smaller atom components that are needed I think this is a great start. 
What we will probably want to firm up before we dive into v5 is the proper way to submit new components, ie starting with a PR that just describes the component with some screenshots. I think a solid process will keep the library cleaner and more concise

**Comment by thomasthesecond** [comment]
Yeah for sure. I definitely want to put rules in place to help define when something should be added and what it should have when added (i.e., test coverage, readme)

**Comment by shawncothran** [comment]
Should the larger molecules and/or organisms, more frequently reused complex components, be kept elsewhere in backpack , or otherwise, as an additional package(s)? I think you are right in that there really aren't very many of these larger components that are really reused more than a couple times.

**Comment by thomasthesecond** [comment]
@shawncothran I think those type of components you mention could be placed along site atom-type components in backpack with whatever approach we take in #617.
